Userland/Utilities/uniq.cpp

```diff
@@ -5,75 +5,124 @@
+            else if (!in_field) {
+                in_field = true;
+                field_index = i;
+                if (++current_field > field_skip_count)
+                    break;
+            }
+        }
+        line = line.substring_view(field_index);
     }
+    char_skip_count = min(char_skip_count, line.length());
+    return line.substring_view(char_skip_count);
+}
 
-    return ret;
+// Borrowed from cmp.cpp, there should probably be a simpler way
+static ErrorOr<NonnullOwnPtr<Core::Stream::BufferedFile>> open_file_or_stdin(String const& filename)
+{
+    OwnPtr<Core::Stream::File> file;
+    if (filename.is_empty() || filename == "-")
+        file = TRY(Core::Stream::File::adopt_fd(STDIN_FILENO, Core::Stream::OpenMode::Read));
+    else
+        file = TRY(Core::Stream::File::open(filename, Core::Stream::OpenMode::Read));
+    return TRY(Core::Stream::BufferedFile::create(file.release_nonnull()));
 }
 
 ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
     TRY(Core::System::pledge("stdio rpath wpath cpath"));
 
-    char const* inpath = nullptr;
-    char const* outpath = nullptr;
+    StringView inpath;
+    StringView outpath;
+    bool duplicates = false;
+    bool unique = false;
+    bool ignore_case = false;
+    bool print_count = false;
+    unsigned int skip_chars = 0;
+    unsigned int skip_fields = 0;
+
     Core::ArgsParser args_parser;
+    args_parser.add_option(duplicates, "Only print duplicated lines", "repeated", 'd');
+    args_parser.add_option(unique, "Only print unique lines (default)", "unique", 'u');
+    args_parser.add_option(ignore_case, "Ignore case when comparing lines", "ignore-case", 'i');
+    args_parser.add_option(print_count, "Prefix each line by its number of occurences", "count", 'c');
+    args_parser.add_option(skip_chars, "Skip N chars", "skip-chars", 's', "N");
+    args_parser.add_option(skip_fields, "Skip N fields", "skip-fields", 'f', "N");
     args_parser.add_positional_argument(inpath, "Input file", "input", Core::ArgsParser::Required::No);
     args_parser.add_positional_argument(outpath, "Output file", "output", Core::ArgsParser::Required::No);
     args_parser.parse(arguments);
 
-    FILE* infile = get_stream(inpath, "r");
-    FILE* outfile = get_stream(outpath, "w");
-
-    struct linebuf buffers[2];
-    struct linebuf* previous = &(buffers[0]);
-    struct linebuf* current = &(buffers[1]);
-    bool first_run = true;
-    for (;;) {
-        errno = 0;
-        ssize_t rc = getline(&(current->buf), &(current->len), infile);
-        if (rc < 0 && errno != 0) {
-            perror("getline");
-            exit(1);
-        }
-        if (rc < 0)
-            break;
-        if (!first_run && strcmp(current->buf, previous->buf) == 0)
-            continue;
+    if (!unique && !duplicates)
+        unique = true;
+    else if (unique && duplicates)
+        // Printing duplicated and unique lines shouldn't print anything
+        exit(0);
```

Comment:
Nitpick: I'd probably add the braces around both branches of this `if` since the comment makes the `exit(0)` belonging to the `else` branch slightly harder to parse. 

Also I think you can just `return {};`  here rather than `exit(0);` this this is main.

Reply:
`ErrorOr<int>`, so I gotta `return 0`. :^)